iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -150,7 +150,7 @@ private static void writeExtensionNamespaces(Network n, ExportOptions options, X
 
     private static void writeExtension(Extension<? extends Identifiable<?>> extension, NetworkXmlWriterContext context) throws XMLStreamException {
         XMLStreamWriter writer = context.getExtensionsWriter();
-        ExtensionXmlSerializer extensionXmlSerializer = getExtensionXmlSerializer(context.getOptions(), extension);
+        ExtensionXmlSerializer extensionXmlSerializer = Objects.requireNonNull(getExtensionXmlSerializer(context.getOptions(), extension));
```

Comment:
`Objects.requireNonNull` throws a NullPointerException: do we need a more specific exception like it's thrown in getExtensionXmlSerializer?

I think we never have a null extensionXmlSerializer there. The `getExtensionXmlSerializer` return null in two cases:
- the extension should not be serialized (no data for the current variant). It never happens because these extensions are filtered in `writeExtensions`
- the extension is skipped (options). It never happens because these extensions are filtered in `writeExtensions`

From my point of view, we should throw an AssertionError here with an explicit message if the serializer is null because something really unexpected happened

Reply:
As said in the comment of the PR, extensionXmlSerializer is **never** null because when `writeExtension` is called, it has already been checked that `extension` has an XML serializer. I added a `requireNonNull` to reinforce that when we go here, we are not supposed to have a null XML serializer. What bothers me is that `getExtensionXmlSerializer` twice though.